File: rust/src/models/endpoint_disabled_event.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::endpoint_disabled_event_data::EndpointDisabledEventData;

/// Sent when an endpoint has been automatically disabled after continuous
/// failures, or manually via an API call.
#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct EndpointDisabledEvent {
    pub data: EndpointDisabledEventData,

    pub r#type: String,
}

impl EndpointDisabledEvent {
    pub fn new(data: EndpointDisabledEventData, r#type: String) -> Self {
        Self { data, r#type }
    }
}

File: javascript/src/models/endpointDisabledEvent.ts
// this file is @generated
import {
  type EndpointDisabledEventData,
  EndpointDisabledEventDataSerializer,
} from "./endpointDisabledEventData";

/** Sent when an endpoint has been automatically disabled after continuous failures, or manually via an API call. */
export interface EndpointDisabledEvent {
  data: EndpointDisabledEventData;
  type: string;
}

export const EndpointDisabledEventSerializer = {
  _fromJsonObject(object: any): EndpointDisabledEvent {
    return {
      data: EndpointDisabledEventDataSerializer._fromJsonObject(object["data"]),
      type: object["type"],
    };
  },

  _toJsonObject(self: EndpointDisabledEvent): any {
    return {
      data: EndpointDisabledEventDataSerializer._toJsonObject(self.data),
      type: self.type,
    };
  },
};
